apps/fly-my-rockets/src/app/rockets/dialogs/flight-dialog/flight-dialog.component.ts

```diff
@@ -0,0 +1,57 @@
+import { Component, Inject } from '@angular/core';
+import { FormArray, FormBuilder, FormControl, FormGroup, Validators } from '@angular/forms';
+import { MAT_DIALOG_DATA, MatDialogRef } from '@angular/material/dialog';
+import { Flight } from '../../rocket.model';
+
+@Component({
+  selector: 'fmr-flight-dialog',
+  templateUrl: './flight-dialog.component.html',
+  styles: [
+    `
+      mat-form-field {
+        display: block;
+      }
+
+      .float-right {
+        float: right;
+      }
+    `
+  ]
+})
+export class FlightDialogComponent {
+  flightForm: FormGroup;
+
+  constructor(
+    public dialogRef: MatDialogRef<FlightDialogComponent>,
+    @Inject(MAT_DIALOG_DATA) public data: Flight,
+    fb: FormBuilder
+  ) {
+    this.flightForm = fb.group({
+      date: [data.date?.toDate(), Validators.required],
```

Comment:
`date` comes from Firebase as an object with seconds and nanoseconds. Turn it into a regular date by calling `.toDate()` on it before feeding it to the form.